nexus/transport.go

```diff
@@ -78,6 +80,43 @@ type Transport interface {
 	//
 	// NOTE: Experimental
 	Close() error
+
+	// UnimplementedTransport must be embedded into any [Transport] implementation for future compatibility.
+	// It implements all methods on the [Transport] interface, returning unimplemented errors if they are not implemented by
+	// the embedding type.
+	mustEmbedUnimplementedTransport()
+}
+
+// TransportError indicates a client encountered something unexpected in the server's response.
+type TransportError struct {
+	// Error message.
+	Message string
+	// Optional failure that may have been embedded in the response.
+	Failure *Failure
+	// Additional transport specific details.
+	// For HTTP, this would include the HTTP response. The response body will have already been read into memory and
+	// does not need to be closed.
+	Details any
```

Comment:
Should we change this to `Cause error`? That would allow unwrapping to get the underlying error.

We could make a `struct HTTPResponseError { *http.Response }` helper for that.